File: admin-ui/lib/api/types/telegram/TelegramChatDto.ts
export type TelegramChatDto = {
    id: number;
    type: TelegramChatDto.Type;
    title: string | null;
    username: string | null;
    firstName: string | null;
    lastName: string | null;
    smallPhotoUrl: string | null;
    bigPhotoUrl: string | null;
    bio: string | null;
    description: string | null;
    inviteLink: string | null;
    linkedChatId: string | null;
};

export namespace TelegramChatDto {
    export type Type = "PRIVATE" | "GROUP" | "SUPERGROUP" | "CHANNEL";
}

export type TelegramChatFetchResultDto = {
    found: boolean;
    chat: TelegramChatDto | null;
}

File: src/main/kotlin/space/davids_digital/kiri/rest/dto/telegram/TelegramChatDto.kt
package space.davids_digital.kiri.rest.dto.telegram

data class TelegramChatDto (
    val id: Long,
    val type: Type,
    val title: String? = null,
    val username: String? = null,
    val firstName: String? = null,
    val lastName: String? = null,
    // TODO
    // val smallPhotoUrl: String? = null,
    // val bigPhotoUrl: String? = null,
    val bio: String? = null,
    val description: String? = null,
    val inviteLink: String? = null,
    val linkedChatId: Long? = null,
) {
    enum class Type {
        PRIVATE,
        GROUP,
        SUPERGROUP,
        CHANNEL
    }
}